### PR TITLE
flux-accounting guide: add citation to Slurm

### LINKED ISSUE #160: accounting: add citation to Slurm accounting
While working on #150, a good suggestion was made to add a brief citation to explain the need for adopting similar terms to Slurm's open source accounting docs in order to meet local management requirements. This should be added to the flux-accounting section in the docs site.

### PR DESCRIPTION
**Pull request by cmoussa1**
#### Background

While working on #150, a suggestion was made to add a brief citation to explain the need for adopting similar terms to Slurm's open source accounting docs in order to meet local management requirements.

---

This PR adds a notice about the similarities in both terms and concepts between flux-accounting and Slurm at the bottom of the guide, with links to both the Slurm documentation as well as the GNU General Public License documentation.

Fixes #160

### CODE REVIEW

File: flux-accounting-guide.rst [hunk 1]
```diff
@@ -148,4 +148,17 @@ association
...
+
+ Much of the terminology and concepts present in flux-accounting is
+ similar to that of `Slurm`_. The reason for this is to meet and maintain local
+ management requirements, as both were developed at LLNL. The terms and
+ concepts adopted from Slurm are taken with guidance from the `GNU General
+ Public License`_.
+
 .. _documentation: https://sqlite.org/omitted.html
+
+.. _Slurm: https://slurm.schedmd.com/documentation.html
```
Comment (garlick) [inline flux-accounting-guide.rst:162]: Rewording suggestion:

> The design of flux-accounting was driven by LLNL site requirements.  Years ago, the design of the Slurm accounting (citation), and its multi-factor priority plugin (citation)  were driven by similar LLNL site requirements.    We chose to reuse terminology and concepts from Slurm to facilitate a smooth transition to Flux.  The flux-accounting code base is all completely new, however.

https://slurm.schedmd.com/accounting.html

https://slurm.schedmd.com/priority_multifactor.html

Comment (cmoussa1) [reply flux-accounting-guide.rst]: Thanks for this suggestion! I just pushed up a fix that includes your rewording recommendation.

**Review by garlick — APPROVED** [review]
Thanks!  Your version was fine, but I thought the direct slurm doc references might be useful for context, and the mere mention of licensing seems to invite an off-topic conversation with some people :stuck_out_tongue_closed_eyes: so perhaps it was best omitted.

**Comment by cmoussa1** [comment]
> the mere mention of licensing seems to invite an off-topic conversation with some people 😝 so perhaps it was best omitted.

That makes total sense. I hadn't thought of that, so I appreciate you pointing it out! I'll set MWP on this. Thanks for reviewing and approving @garlick!

**Pull request merged**